### PR TITLE
Add counters to Metric

### PR DESCRIPTION
**Pull request by Montagon**
This PR updates the Metric capabilities by adding the possibility of adding counters. A new interface (`CounterMetric`) and implementations for `LogsMetric` and `OpenTelemtry` have been added to support this.

The docker-compose files and configuration files for OpenTelemtry have also been moved inside the OpenTelemetry module.

### CODE REVIEW

**Review by javipacheco — APPROVED** [review]
🚀

File: integrations/opentelemetry/src/main/kotlin/com/xebia/functional/xef/opentelemetry/OpenTelemetryCounter.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+class OpenTelemetryCounter(private val longCounter: io.opentelemetry.api.metrics.LongCounter) :
+  CounterMetric {
+  override fun increment(n: Long) {
+    longCounter.add(n)
+  }
+
+  override fun decrement(n: Long) {
+    longCounter.add(n)
+  }
```
Comment (franciscodr) [inline integrations/opentelemetry/src/main/kotlin/com/xebia/functional/xef/opentelemetry/OpenTelemetryCounter.kt:13]: Why do we call the' add' method if the function's purpose is decrementing the counter?

File: examples/src/main/kotlin/com/xebia/functional/xef/assistants/DSL.kt [hunk 2]
```diff
@@ -42,7 +42,9 @@ suspend fun main() {
   //  - # docker-compose up
 
   val metric = com.xebia.functional.xef.metrics.Metric.EMPTY
-  // val metric = com.xebia.functional.xef.opentelemetry.OpenTelemetryMetric()
+  //    val metric = com.xebia.functional.xef.opentelemetry.OpenTelemetryMetric()
+  //  val metric = com.xebia.functional.xef.metrics.LogsMetric()
```
Comment (franciscodr) [inline examples/src/main/kotlin/com/xebia/functional/xef/assistants/DSL.kt:46]:
```suggestion
```

**Review by franciscodr — APPROVED** [review]

File: integrations/opentelemetry/src/main/kotlin/com/xebia/functional/xef/opentelemetry/OpenTelemetryCounter.kt (hunk 1 above)
Comment (Montagon) [reply integrations/opentelemetry/src/main/kotlin/com/xebia/functional/xef/opentelemetry/OpenTelemetryCounter.kt]: Removed, thanks!

**Review by franciscodr — APPROVED** [review]

**Pull request merged**